### PR TITLE
Add some checks for HadoopTables#create

### PR DESCRIPTION
**Pull request by chenjunjiedada**

### CODE REVIEW

**Comment by chenjunjiedada** [comment]
close and reopen to trigger CI.

File: core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java [hunk 1]
```diff
@@ -70,18 +72,23 @@ public Table load(String location) {
    *
    * @param schema iceberg schema used to create the table
    * @param spec partition specification
+   * @param properties properties of the table to be created
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java]: Can you note that null is accepted?

Also, should we do something similar for spec? If spec is null, we could use `PartitionSpec.unpartitioned()`.

Comment (chenjunjiedada) [reply core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java]: Sure, please see the latest code change.

**Comment by chenjunjiedada** [comment]
Recently, some unit tests fail randomly. Close and reopen to trigger CI again. We need to fix that.

File: core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java [hunk 2]
```diff
@@ -69,19 +71,24 @@ public Table load(String location) {
    * location.
    *
    * @param schema iceberg schema used to create the table
-   * @param spec partition specification
+   * @param spec partition specification. It can be null in case of unpartitioned table
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java]: How about "partitioning spec, if null the table will be unpartitioned"

File: core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java [hunk 3]
```diff
@@ -69,19 +71,24 @@ public Table load(String location) {
...
+   * @param properties properties of the table to be created, it can be null
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java]: How about "a string map of table properties, initialized to empty if null"

Comment (chenjunjiedada) [reply core/src/main/java/org/apache/iceberg/hadoop/HadoopTables.java]: Done

**Pull request merged**